Garrett-Iannuzzi/palette-picker
# Set up migrations

#### What does this PR do?
- Adds initial setup/instals for knex, express, postgres, etc...
- Adds `GET` request for all projects
- Adds `GET` request for a palette by its ID
- Adds `GET` request for a project by ID
- Adds `GET` request for all palettes by an ID from a project
- Adds `POST` request for adding a project 
- Adds `POST` request for adding a palette 
- Adds `PATCH` request for adding a palette to a project
- Adds `PATCH` request for  changing the name of a project 
- Adds `DELETE` request for removing a project by its ID
- Adds `DELETE` request for removing a palette by its ID

#### How to test?
- Pull down `set-up-migrations` branch 
- Make sure you have the above tech installed 
- For now you test all endpoints in postman to make sure the correct status codes are returned

#### Issues:
- No issues have been closed
- The project board has been updated 
- One issue was added, here is a screenshot of that code:

<img width="765" alt="Screen Shot 2020-02-04 at 3 57 14 PM" src="https://user-images.githubusercontent.com/48968224/73795084-063bb000-4767-11ea-8212-f5460dbe2995.png">

#### Notes:
- Nothing yet
- Next up: I will work on updating the README with the endpoints 

Thanks!
@adamsjr8576
